File: array/array9.go
package array

func twoSum(nums []int, target int) []int {
	check := map[int]int{}
	// [3,3]
	for i, v := range nums {

		gap := target - v
		value, isExist := check[gap]
		if isExist {
			return []int{value, i}
		} else {
			check[v] = i
		}

	}
	return nums
}

File: old/Array/array9.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let memo = {};
  let result = [];
  for (let i = 0; i < nums.length; i++) {
    let value = nums[i];
    let gap = target - value;
    if (memo[gap] !== undefined) {
      return [i, memo[gap]];
    } else memo[value] = i; // (key - 값 : value - index) 로 memo
  }
};
